src/Storages/Kafka/KafkaConsumer.cpp

```diff
@@ -590,6 +592,9 @@ KafkaConsumer::Stat KafkaConsumer::getStat() const
         });
     }
 
+    auto [exc_buffer, num_exc] = [&](){std::lock_guard<std::mutex> lock(exception_mutex);
+        return std::make_tuple(this->exceptions_buffer, this->num_exceptions);}();
```

Comment:
it looks unnecessarily complicated after moving from the return statement. Could you please make another code block with `lock` and two variables outside of the block?

Reply:
It might be also good for `rdkafka_stat` just to keep it consistent (but should be a different code block to avoid both mutexes held simultaneously).

Reply:
Was this sort of what you had in mind? https://github.com/ClickHouse/ClickHouse/compare/81fff496e7793bd02d17c2f0915a2e4b528a23bf..5f58e43f1f02a64431b95e3f50720dda8199638f

Reply:
Yes, kind of. Even more readable. I've added the changes